src/NUnitTestExtractor/NUnitTestExtractor.Tests/NUnitTestExtractorTests.cs

```diff
@@ -3,22 +3,309 @@
+
+            Assert.That(process.StandardOutput.ReadToEnd(), Is.Not.Empty);
+
+            process.WaitForExit();
+        }
+
+        [TestCaseSource(typeof(InvalidDataTestCases))]
+        public void WritingToStdout_InvalidDllAndEveryLevel_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level);
+
+            Assert.That(process.StandardOutput.ReadToEnd(), Is.Empty);
+
+            process.WaitForExit();
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Namespace)]
+        public void WritingToStdout_LevelIsNamespace_OnlyNamespaceWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level.ToString());
+
+            string line = process.StandardOutput.ReadLine();
+            string result = line.Substring(line.IndexOf("|") + 2);
+
+            Assert.That(!result.Contains("."));
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Class)]
+        public void WritingToStdout_LevelIsClass_NamespaceAndClassWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level.ToString());
+
+            string line = process.StandardOutput.ReadLine();
+            string result = line.Substring(line.IndexOf("|") + 2);
+
+            //This works as class and namespace are seperated by a period example: namepspace.class
+            Assert.That(result.Contains("."));
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Function)]
+        public void WritingToStdout_LevelIsFunction_NamespaceClassAndFunctionGetWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level.ToString());
+
+            string line = process.StandardOutput.ReadLine();
+            string result = line.Substring(line.IndexOf("|") + 2);
+
+            //Same deal as test above, function and class separated with a period: namespace.class.function
+            Assert.That(result.Count(p => p == '.'), Is.EqualTo(2));
         }
 
+        [TestCaseSource(typeof(ValidDataTestCases))]
+        public void WritingToStdout_ValidDllAndEveryLevel_ContainsNoDuplicates(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            Process process = SetUpProcess(dll, level);
+
+            string line = "";
```

Comment:
The Coding Standard recommends using string.Empty instead of "".
I believe "" creates a new empty string which then needs to be garbage collected, while string.Empty is a pre-created constant.